301d58/code-challenges/objects/challenges-06.test.js

```diff
@@ -0,0 +1,260 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1
+
+Write a function named getCourseKeys that takes in the courseInfo object and returns an array containing the keys for the courseInfo object.
+
+For example: (['name', 'duration', 'topics', 'finalExam']).
+------------------------------------------------------------------------------------------------ */
+
+const courseInfo = { name: 'Code 301', duration: { dayTrack: '4 weeks', eveningTrack: '8 weeks'},
+  topics: ['SMACSS', 'APIs', 'NodeJS', 'SQL', 'jQuery', 'functional programming'],
+  finalExam: true };
+
+const getCourseKeys = (obj) => {
+  let arr = Object.keys(obj);
+  return arr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+Use the characters data below for the remainder of the challenges.
+
+Write a function named getHouses that returns a new array containing the names of all of the houses in the data set.
+------------------------------------------------------------------------------------------------ */
+
+let characters = [
+  {
+    name: 'Eddard',
+    spouse: 'Catelyn',
+    children: ['Robb', 'Sansa', 'Arya', 'Bran', 'Rickon'],
+    house: 'Stark'
+  },
+  {
+    name: 'Jon A.',
+    spouse: 'Lysa',
+    children: ['Robin'],
+    house: 'Arryn'
+  },
+  {
+    name: 'Cersei',
+    spouse: 'Robert',
+    children: ['Joffrey', 'Myrcella', 'Tommen'],
+    house: 'Lannister'
+  },
+  {
+    name: 'Daenarys',
+    spouse: 'Khal Drogo',
+    children: ['Drogon', 'Rhaegal', 'Viserion'],
+    house: 'Targaryen'
+  },
+  {
+    name: 'Mace',
+    spouse: 'Alerie',
+    children: ['Margaery', 'Loras'],
+    house: 'Tyrell'
+  },
+  {
+    name: 'Sansa',
+    spouse: 'Tyrion',
+    children: [],
+    house: 'Stark'
+  },
+  {
+    name: 'Jon S.',
+    spouse: null,
+    children: [],
+    house: 'Snow'
+  }
+];
+
+const getHouses = (arr) => {
+  let houses = [];
+  arr.forEach( element => {
+    houses.push(element.house);
+  });
+  return houses;
+};
+
+/*------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named hasChildrenValues that uses Object.values to determine if any given character in the data set has children.
+
+This function should take in an array of data and a character name and return a Boolean.
+
+For example:
+hasChildrenValues(characters, 'Cersei') will return true
+hasChildrenValues(characters, 'Sansa') will return false
+------------------------------------------------------------------------------------------------ */
+
+let hasChildrenValues = (arr, character) => {
+  let hasValue = true;
+  Object.values(arr).forEach( element =>{
```

Comment:
`Object.values()` needs to take in an object, here you've passed in an array. Because of this, I'm not sure `Object.values()` is actually doing "it's job" and is instead just returning the array.

`Object.values(arr)` and `arr` might be the same here, if you care to see if I'm right try them both out in the console.

That being said, this does pass the tests and uses the method you were supposed to so good job!